src/ImmersedBoundaries/partial_cell_bottom.jl

```diff
@@ -63,13 +63,40 @@ end
 function ImmersedBoundaryGrid(grid, ib::PartialCellBottom)
     bottom_field = Field{Center, Center, Nothing}(grid)
     set!(bottom_field, ib.bottom_height)
-    @apply_regionally clamp_bottom_height!(bottom_field, grid)
+    @apply_regionally compute_numerical_bottom_height!(bottom_field, grid, ib)
     fill_halo_regions!(bottom_field)
     new_ib = PartialCellBottom(bottom_field, ib.minimum_fractional_cell_height)
     TX, TY, TZ = topology(grid)
     return ImmersedBoundaryGrid{TX, TY, TZ}(grid, new_ib)
 end
 
+@kernel function _compute_numerical_bottom_height!(bottom_field, grid, ib::PartialCellBottom)
+    i, j = @index(Global, NTuple)
+
+    # Save analytical bottom height
+    zb = @inbounds bottom_field[i, j, 1]
+
+    # Cap bottom height at Lz and at rnode(i, j, grid.Nz+1, grid, c, c, f)
+    domain_bottom = znode(i, j, 1, grid, c, c, f)
+    domain_top    = znode(i, j, grid.Nz+1, grid, c, c, f)
+    @inbounds bottom_field[i, j, 1] = clamp(zb, domain_bottom, domain_top)
+
+    ϵ  = ib.minimum_fractional_cell_height
+    for k in 1:grid.Nz
+        z⁻ = znode(i, j, k,   grid, c, c, f)
+        z⁺ = znode(i, j, k+1, grid, c, c, f)
+        Δz = Δzᶜᶜᶜ(i, j, k, grid)
+        bottom_cell = (z⁻ ≤ zb) & (z⁺ ≥ zb)
+
+        # If the size of the bottom cell is less than ϵ Δz, use 
+        # a simple `GridFittedBottom` approach where the bottom 
+        # height is the top interface of cell k.
```

Comment:
Which is a fine proposal but I think you should change the title of the PR. Imagine us looking for what is causing this kind of different in the future, combing through PRs. It's pretty hidden in this one

Reply:
Hmmm, I thought there was supposed to be no difference. 
Here, the `bottom_height` is set to the actual bottom height if it is smaller than `z⁻ + ϵ Δz`, otherwise the cell is closed. 
So, the only difference with main will be for bottom heights that are in between `z⁻ + ϵ Δz` and `z⁺` (where `z⁻` and `z⁺` are the interfaces of the bottom cell.
I will check the difference in the internal_tide example

Reply:
But I thought before we would keep the cell open (rather than closing it).

Actually I think that MITgcm uses some criteria to decide which one to do --- either snapping up or down depending on which change is least.

Reply:
ah, ok, I understand; open till `ϵ Δz` instead of snapping up.

Reply:
Actually I think we used to use that criteria too. It's something like 

```julia
partial_Δz = z⁺ - zb
too_thin = partial_Δz < ϵ * Δz
snap_up = partial_Δz < ϵ * Δz / 2
snapped_zb = ifelse(snap_up, z⁺, z⁻ + ϵ * Δz)
numerical_zb  = ifelse(too_thin, snapped_zb, zb)
```